addons/website_slides/static/tests/tours/slides_tour_tools.js

```diff
@@ -87,6 +95,126 @@ var addArticleToSection = function (sectionName, pageName, backend) {
 }];
 };
 
+const fillInFileInput = async (input, name, type, content) => {
+    const blob = await fetch(`data:${type};base64,${content}`).then(res => res.blob());
+    const file = new File([blob], name, { type: type});
+    const dataTransfer = new DataTransfer();
+    dataTransfer.items.add(file);
+    input.files = dataTransfer.files;
+    input.dispatchEvent(new Event('change', { bubbles: true }));
+};
+
+const compareBase64Content = async (url, name, type, expectedContent) => {
+    const blob = await fetch(url).then(res => res.blob());
+    const file = new File([blob], name, { type: type});
+    const actualContent = await utils.getDataURLFromFile(file).then(dataURL => dataURL.split(',', 2)[1]);
+    return expectedContent === actualContent;
+};
+
+/**
+ * Test the upload of an image file as a new content and especially the binary content of the uploaded image.
+ */
+const addImageToSection = (sectionName, pageName, backend) => {
+    return [
+{
+    content: 'eLearning: add content to section',
+    trigger: (backend ? 'iframe ' : '') + 'div.o_wslides_slide_list_category_header:contains("' + sectionName + '") a:contains("Add Content")',
```

Comment:
can you use "fstring" ? (same below)

Reply:
yes good idea. I have also stored the “backend ? 'iframe ' : ''” in a constant to use in the fstring to avoid the repetition.